README.md

```diff
@@ -86,3 +86,34 @@ frontend/model/challenges.py
 frontend/routes/api/challenges.py
 frontend/api.yaml # This is a specification of the challenges API (file above) using OpenAPI.
 ```
+### Sprint 3
+During our third sprint, we put all the components of our project together and got them functioning. Our challenge team
+made further progress on their challenge sets. We had enough cipher challenges to convert it ot a challenge set of its
+own and make the remaining challenges a second challenge set which we named web-based as all the challenges there are
+targeted towards teaching the user how a directory traversal or similar web-based attack works.
+
+For the UI we managed to implement a 3rd party library that converts markdown into HTML for being able to display
+documents that are more easily written to the screen as an HTML object. Additionally, we implemented a special useFetch
+call in React that creates isolated environments based on the user token, challenge-set and challenge slugs. Finally,
+to ensure that the user was unable to generate an environment without being logged in, we prevented anything from being
+accessible until the user is logged in.
+
+For the backend, we created the REST APIs that the UI talks with. Currently, the UI does not fully utilize these APIs
+yet, but does use some of them, such as the creation of the user-isolated environments. There is a branch of the project
+that we have already started working on to utilize these APIs to dynamically create the challenges, including
+the documentation required for each.
```

Comment:
Maybe make a mention here of the APIs utilizing access tokens issued by Auth0 to protect sensitive information.